### DISCUSSION TITLE
NFT image not visible

### DISCUSSION THREAD

**Discussion by SuryaSundarVadali** [Cyfrin/foundry-full-course-cu#2192] [Q&A]
![1](https://github.com/user-attachments/assets/a52b695d-00a3-4ac8-9cd7-46262f285f1b)

Image of nft not visible

**Comment by EngrPips** [comment]
Hello @SuryaSundarVadali, Please read this discussion https://github.com/Cyfrin/foundry-full-course-cu/discussions/2112#discussion-6978415 and it should help with having your image displayed

**Reply by SuryaSundarVadali** [reply]
It is still not visible

**Reply by SuryaSundarVadali** [reply]
![2](https://github.com/user-attachments/assets/b41fb4eb-5d88-42d6-8806-f166dbaf05d0)

**Reply by EngrPips** [reply]
That weird, Can you try to view it on Opensea, and let's see if it will be visible there.

**Reply by SuryaSundarVadali** [reply]
It is not visible on opensea as well

**Reply by EngrPips** [reply]
That weird, are you sure token URI json is pointing to the appropriate ipfs link?

**Reply by SuryaSundarVadali** [reply]
I ll check once again and try

**Reply by EngrPips** [reply]
Alright, Let see how it plays out

**Reply by SuryaSundarVadali** [reply]
It is still not working once can you check my code if there is any mistake

**Comment by SuryaSundarVadali** [comment]
`
{
    "name" : "TigerTheCat",
    "description" : "Tiger is a cat",
    "image" : "https://ipfs.io/ipfs/QmfMv8VM1usgFwopBc8Wq2XgCTwE8PZEobvk343wjoZTy9?filename=Tiger.png",
    "attributes": [
        {
            "trait_type": "Sass",
            "value": 100
        }
    ]
}
`
Json file

**Reply by ZxxWzzz** [reply]
Is your tokenURI return value correct?Can you provide the contract address

**Reply by EngrPips** [reply]
@SuryaSundarVadali please check that your tokenURI value is correct

**Comment by SuryaSundarVadali** [comment]
`
contract MintBasicNFT is Script {
    string public constant CAT = "https://ipfs.io/ipfs/QmZc8jCTV2AxFUHZeRAgBdzgT8fB1AVhMxViCUyF2wHtn3?filename=Tiger.json";

    function run() external {
        address mostRecentlyDeployed = DevOpsTools.get_most_recent_deployment(
            "BasicNFT",
            block.chainid
            );
            mintNFTOnContract(mostRecentlyDeployed);
            
    }

    function mintNFTOnContract (address contractAddress) public {
                vm.startBroadcast();
                BasicNFT(contractAddress).mintNFT(CAT);
                vm.stopBroadcast();
            }

}
`
Interactions.s.sol

**Reply by EngrPips** [reply]
I am not sure why the image is not showing up, as the ipfs looks fine to me. I will say continue with the course and you can come back to this later.

**Reply by SuryaSundarVadali** [reply]
Ok thanks for your help!
 i have also asked a metamask developer on reddit for help hopefully he will be able to resolve it

**Reply by EngrPips** [reply]
Let's look forward to him helping resolve it. If he does help you resolve this issue, please let us know the solution.

**Reply by SuryaSundarVadali** [reply]
Sure thanks for your help

**Reply by EngrPips** [reply]
You are welcome, Friend.

**Comment by SuryaSundarVadali** [comment]
Now it is showing like this 
![sd](https://github.com/user-attachments/assets/2e1484a1-6e69-471a-9aae-e702b6598e75)

**Reply by EngrPips** [reply]
After you adjusted the `Metamask` settings so that Metamask could display the image, Did you re-import the NFT?

**Reply by SuryaSundarVadali** [reply]
Yes

**Reply by EngrPips** [reply]
It's weird that it still didn't work for you. It worked for the other guy,

**Reply by vishalsingh2972** [reply]
hey @SuryaSundarVadali is it showing now, i am facing something similar NFT image showing in opensea but NFT image is not showing in metamask wallet.

**Reply by EngrPips** [reply]
If I can view it in Opensea, I wouldn't worry too much about metamask.
